iati/core/tests/test_data.py

```diff
@@ -197,32 +212,43 @@ def test_instantiation_dataset_from_string_with_encoding(self, encoding):
     @pytest.mark.parametrize("encoding_declared, encoding_used", [
         ("UTF-16", "UTF-8"),
         ("UTF-16", "ISO-8859-1"),
+        ("UTF-16", "ASCII"),
         ("UTF-16", "BIG5"),
-        ("UTF-16", "EUC-JP"),
-        ("ASCII", "UTF-16"),
-        ("ISO-8859-1", "UTF-16"),
-        ("ISO-8859-2", "UTF-16"),
-        ("BIG5", "UTF-16"),
-        ("EUC-JP", "UTF-16")])
-    def test_instantiation_dataset_from_string_with_encoding_mismatch(self, encoding_declared, encoding_used):
+        ("UTF-16", "EUC-JP")
```

Comment:
See the `Note` in the docstring / the commit message in https://github.com/IATI/iati.core/pull/95/commits/89fb099e0ddc718c1615066199c51f5ef9d931cf for an explanation of why it's the way it is.